.infrastructure/09_iam_role.tf

```diff
@@ -0,0 +1,113 @@
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecs-task-execution-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
```

Comment:
I'm making this comment just to clarify something in my mind: why do you need this `assume_role_polocy` if you then attach the policy `"arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"` to the resource?

Reply:
"assume_role_policy"  is a mandatory attribute to be set when creating a new role (ecs-task-execution-role) and is used to define which entity (in this case: "Service": "ecs-tasks.amazonaws.com") can assume that role. The policy "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy" (and other) is the AWS Managed policy which contains the permissions for the common use cases for ECS and is attached to the role defined previously (ecs-task-execution-role).